core/Psr4Autoloader.php

```diff
@@ -82,48 +84,46 @@ public function prefixes($prefix = '')
      */
     public function register()
     {
-        spl_autoload_register(array($this, 'loadClass'));
+        spl_autoload_register([$this, 'loadClass']);
     }
 
 
     /**
      * Adds a base directory for a namespace prefix.
      *
-     * @param string $prefix   The namespace prefix.
-     * @param string $base_dir A base directory for class files in the
-     *                         namespace.
-     * @param bool   $prepend  If true, prepend the base directory to the stack
-     *                         instead of appending it; this causes it to be searched first rather
-     *                         than last.
-     * @return void
+     * @param string $prefix    The namespace prefix.
+     * @param string $base_dir  A base directory for class files in the namespace.
+     * @param bool $prepend     If true, prepend the base directory to the stack instead of appending it;
+     *                          this causes it to be searched first rather than last.
+     * @return bool             returns TRUE if the namespace was successfully added
      */
-    public function addNamespace($prefix, $base_dir, $prepend = false)
+    public function addNamespace(string $prefix, string $base_dir, bool $prepend = false): bool
     {
         // normalize namespace prefix
         $prefix = trim($prefix, Psr4Autoloader::NS) . Psr4Autoloader::NS;
         // normalize the base directory with a trailing separator
-        $base_dir = \EEH_File::standardise_and_end_with_directory_separator($base_dir);
+        $base_dir = EEH_File::standardise_and_end_with_directory_separator($base_dir);
         // initialize the namespace prefix array
-        if (isset($this->prefixes[ $prefix ]) === false) {
-            $this->prefixes[ $prefix ] = array();
+        if (! isset($this->prefixes[ $prefix ])) {
+            $this->prefixes[ $prefix ] = [];
```

Comment:
Is it possible that `false` may be a special value as compared to empty strings? May be `$prefix` can be an empty string?

Reply:
it wasn't checking if `$prefix` or `$this->prefixes[ $prefix ]` were equal to false, it was checking the result of calling `isset()` on `$this->prefixes[ $prefix ]`, and that function will return a bool.